### ISSUE TITLE
Onresume not fires and Workflow remains suspended

### ISSUE THREAD

**Issue by osmankoc** [elsa-workflows/elsa-core#4002]
Hi,

I created a solution based on the tutorial on https://medium.com/codex/part-6-of-building-workflow-driven-net-applications-with-elsa-2-9b3167c612dd
.

At the Updateblockchain step workflow remains suspended. 

Source code is 
[DocumentManagement.zip](https://github.com/elsa-workflows/elsa-core/files/11405913/DocumentManagement.zip)

Thanks..

`using System;
using System.IO;
using System.Security.Cryptography;
using System.Threading;
using System.Threading.Tasks;
using Elsa;
using Elsa.ActivityResults;
using Elsa.Attributes;
using Elsa.Expressions;
using Elsa.Models;
using Elsa.Providers.WorkflowStorage;
using Elsa.Services;
using Elsa.Services.Models;
using Hangfire;

namespace DocumentManagement.Workflows.Activities
{
    [Activity(
        Category = "Document Management",
        Description = "Saves a hash of the specified file onto the blockchain to prevent tampering."
    )]
    public class UpdateBlockchain : Activity
    {
        public record UpdateBlockchainContext(string WorkflowInstanceId, string ActivityId, string FileSignature);

        private readonly IBackgroundJobClient _backgroundJobClient;
        private readonly IWorkflowInstanceDispatcher _workflowInstanceDispatcher;

        public UpdateBlockchain(IBackgroundJobClient backgroundJobClient, IWorkflowInstanceDispatcher workflowInstanceDispatcher)
        {
            _backgroundJobClient = backgroundJobClient;
            _workflowInstanceDispatcher = workflowInstanceDispatcher;
        }

        [ActivityInput(
            Label = "File",
            Hint = "The file to store its hash of onto the blockchain. Can be byte[] or Stream.",
            SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid },
            DefaultWorkflowStorageProvider = TransientWorkflowStorageProvider.ProviderName
        )]
        public object File { get; set; } = default!;

        [ActivityOutput(Hint = "The computed file signature as stored on the blockchain.")]
        public string Output { get; set; } = default!;

        /// <summary>
        /// Invoked by Hangfire as a background job.
        /// </summary>
        public async Task SubmitToBlockChainAsync(UpdateBlockchainContext context, CancellationToken cancellationToken)
        {
            // Simulate storing it on an imaginary blockchain out there.
            await Task.Delay(TimeSpan.FromSeconds(15), cancellationToken);

            // Resume the suspended workflow.
            await _workflowInstanceDispatcher.DispatchAsync(new ExecuteWorkflowInstanceRequest(context.WorkflowInstanceId, context.ActivityId, new WorkflowInput(context.FileSignature)), cancellationToken);
        }

        protected override async ValueTask<IActivityExecutionResult> OnExecuteAsync(ActivityExecutionContext context)
        {
            // Compute hash.
            var bytes = File is Stream stream ? await stream.ReadBytesToEndAsync() : File is byte[] buffer ? buffer : throw new NotSupportedException();
            var fileSignature = ComputeSignature(bytes);

            // Schedule background work using Hangfire.
            _backgroundJobClient.Enqueue(() => SubmitToBlockChainAsync(new UpdateBlockchainContext(context.WorkflowInstance.Id, context.ActivityId, fileSignature), CancellationToken.None));

            // Suspend the workflow.
            return Suspend();
        }

        protected override IActivityExecutionResult OnResume(ActivityExecutionContext context)
        {
            // When we resume, simply complete this activity.
            var fileSignature = context.GetInput<string>();
            Output = fileSignature;
            return Done();
        }

        private static string ComputeSignature(byte[] bytes)
        {
            using var algorithm = SHA256.Create();
            var hashValue = algorithm.ComputeHash(bytes);
            return Convert.ToBase64String(hashValue);
        }
    }
}`

**Comment by osmankoc** [comment]
Any help? actişvity never resumes and instance remains suspended :(

**Comment by sfmskywalker** [comment]
I'll take a look as soon as I can, this issue is on my shortlist for this week.

**Comment by osmankoc** [comment]
Hi. Any progress on this?

**Comment by sfmskywalker** [comment]
Hi! I downloaded your zip file and ran the solution, and I was able to run the workflow without it remaining suspended. In other words, the workflow ran from start to end.

Keep in mind, though, that the `IndexBlockchain` activity emulates a long-running job by waiting for 15 seconds, before resuming execution.

**Comment by osmankoc** [comment]
But i Didn't get the same result. What am i doing wrong?

**Comment by sfmskywalker** [comment]
I don't know, but I can show you what I did:

https://github.com/elsa-workflows/elsa-core/assets/938393/d26681c3-e809-4581-bf08-0245f43ccf16

**Comment by osmankoc** [comment]
Unfortunately i can't see the video :(

**Comment by osmankoc** [comment]
Ok. I saw. Bu it simply not works on my development machine. I also tried it on my coworkers' dev machine either. We use vs 2022. .net core 6.0. 

You didn't make any library or package changes on the solution right?

**Comment by sfmskywalker** [comment]
No changes were made - I literally took your zip file and run it on port 11000 and connected the designer to it. No package references were updated.

**Comment by sfmskywalker** [comment]
Yes, try and debug what’s going on. The easiest way to do so is to clone the Elsa source code and reference the projects directly instead of the packages, this way you can step through the source code.

Unfortunately there’s little I can do if I can’t reproduce the problem.

**Comment by osmankoc** [comment]
Thanks. I'll try that and if i find the problem i'll give you update from here.

**Comment by sfmskywalker** [comment]
Sounds good, thanks!

**Comment by sfmskywalker** [comment]
Closing because this issue targets Elsa 2 or the legacy dashboard stack, which is no longer an actively supported target in this repository. Current UI work happens in Elsa Studio for Elsa 3. If you can still reproduce the underlying problem on a supported Elsa 3 + Elsa Studio setup, please open a new issue with fresh reproduction steps.
